subworkflows/local/prepare_genome/main.nf

```diff
@@ -98,23 +103,73 @@ workflow PREPARE_GENOME {
         ch_exon_bed = ch_exon_bed_raw
     }
 
-    def ch_dbsnp = Channel.value([[], []])
-    def ch_dbsnp_tbi = Channel.value([[], []])
-    if (params.dbsnp && !params.dbsnp.toString().endsWith(".gz")) {
-        BGZIPTABIX_DBSNP(
-            dbsnp
-        )
-        ch_versions = ch_versions.mix(BGZIPTABIX_DBSNP.out.versions)
-        ch_dbsnp = BGZIPTABIX_DBSNP.out.gz_tbi.map { meta, vcf, _tbi -> [meta, vcf] }.collect()
-        ch_dbsnp_tbi = BGZIPTABIX_DBSNP.out.gz_tbi.map { meta, _vcf, tbi -> [meta, tbi] }.collect()
-    }
-    else if (params.dbsnp && !params.dbsnp_tbi) {
-        TABIX_DBSNP(
-            dbsnp
-        )
-        ch_versions = ch_versions.mix(TABIX_DBSNP.out.versions)
-        ch_dbsnp_tbi = TABIX_DBSNP.out.tbi.collect()
-    }
+    def ch_bcftools_annotations_input = bcftools_annotations
+        .map { file -> [[id: file.name], file] }
+        .join(bcftools_annotations_tbi.map { file -> [[id: file.baseName], file] }, failOnDuplicate: true, remainder: true)
+        .branch { meta, file, index ->
+            plain: !file.toString().endsWith(".gz")
+            return [meta, file]
+            bgzip_noindex: !index && file.toString().endsWith(".gz")
+            return [meta, file]
+            bgzip_index: true
+            return [meta, file, index]
+        }
+
+    BGZIPTABIX_BCFTOOLS_ANNOTATIONS(
+        ch_bcftools_annotations_input.plain
+    )
+    ch_versions = ch_versions.mix(BGZIPTABIX_BCFTOOLS_ANNOTATIONS.out.versions)
+
+    TABIX_BCFTOOLS_ANNOTATIONS(
+        ch_bcftools_annotations_input.bgzip_noindex
+    )
+    ch_versions = ch_versions.mix(TABIX_BCFTOOLS_ANNOTATIONS.out.versions)
+
+    def ch_bcftools_annotations = BGZIPTABIX_BCFTOOLS_ANNOTATIONS.out.gz_tbi
```

Comment:
Is it best practise to `def` channel names like this? I thought that was reserved for 'standard' variables

Reply:
I prefer to also `def` channel names. It doesn't hurt anyone and actually gives you a warning if you try to overwrite these in operators